src/spec/payid-protocol.md

```diff
@@ -259,38 +246,41 @@ This specification defines the PayID protocol - an application-layer protocol, w
+### Header PayID-Version
+  The PayID client MAY include the PayID version request header field to specify the version of the PayID protocol used to generate the request.
 
-  If present on a request, the PayID server MUST interpret the request according to the rules defined in the specified version of the PayID protocol or fail the request with an appropriate error response code.
+  If present on a request, the PayID server SHOULD interpret the request according to the rules defined in the specified version of the PayID protocol or fail the request with an appropriate error response code.
 
-  If not specified in a request, the PayID server MUST fail the request with an appropriate error code.
+  If not specified in a request, the PayID server MAY fail the request with an appropriate error code, or may return a PayID response object which includes the `version` key to indicate which version of the PayID Protocol the response payload is following.
 
 ## Request Headers
-  In addition to common Headers, the PayID client MUST specify the following request header.
+  In addition to common Headers, the PayID client MAY specify the following request header.
 
 ### Header Accept
-  The PayID client MUST specify the `Accept` request header field with at least one of the registered media types (Section X). The purpose of this header is to indicate what type of content can be understood in the response. It specifies the `paymentNetwork` and `environment` of the payment account and its representation format for which the PayID client wants to receive information. The representation format is always JSON.
+  The PayID client MAY specify the `Accept` request header field with at least one of the registered media types (Section X). The purpose of this header is to indicate what type of content can be understood in the response. It specifies the `paymentNetwork` and `environment` of the payment account and its representation format for which the PayID client wants to receive information. The representation format is always JSON.
 
-  The PayID server MUST reject formats that specify unknown or unsupported format parameters.
+  The PayID server MAY reject formats that specify unknown or unsupported format parameters.
 
 ## Response Headers
-  In addition to the Common Headers, the PayID server MUST specify the following response header.
+  In addition to the Common Headers, the PayID server SHOULD specify the following response header.
 
 ### Header Cache-Control
-  The PayID server MUST include the `Cache-Control` header with a value `Cache-Control: no-store`. PayID responses MUST NOT be stored in any cache.
+  The PayID server SHOULD include the `Cache-Control` header, to indicate how the PayID server operator wants PayID responses to be cached.
+
+###  Header Access-Control-Allow-Origin
+  The PayID server MUST include a `Access-Control-Allow-Origin: *` header, allowing all origins to make cross-origin requests to resolve PayIDs.
```

Comment:
@sappenin thinks CORS headers should be a SHOULD, not a MUST, but it seems reasonable to require it to allow PayIDs to be accessible from the browser.

Reply:
Agreed - I lean towards `SHOULD` but am not unhappy with `MUST`.